doc/proposals/webhook-proposal.md

```diff
@@ -0,0 +1,137 @@
+        CertDir: hookServerCfg.CertDir,
+    }
+    
+    if err := mgr.Add(hookServer); err != nil {
+        log.Error(err, "Unable to register webhook server")
+        return err
+    }
+
+    log.Info("Registering webhooks to the webhook server")
+    
+    hookServer.Register("/mutate-pods", &webhook.Admission{Handler: &podAnnotator{}})
+    hookServer.Register("/validate-pods", &webhook.Admission{Handler: &podValidator{}})
+    
+    return nil
+}
+```
+
+
+In addition to the above file being used to start up the server, an additional  files (`mutationswebhook.go` and `validationwebhooks.go`) will generated when the user chooses to include webhooks. These are the files which have the necesary boilerplate for a user to implement the `Handle` function in order to run custom logic on validation and admission of pods.
+
+```go
+type podInteraction struct {
+	client  client.Client
+	decoder *admission.Decoder
+}
+
+func (i *podInteraction) Handle(ctx context.Context, req admission.Request) admission.Response {
+	pod := &corev1.Pod{}
+
+	err := i.decoder.Decode(req, pod)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+    //Insert mutation/validation logic here
+
+    //BRANCH
+    //Block A contains the boiler plate logic for patching a mutation
+    //Block B contains the boilder plate logic for 
+
+    //BLOCK A
+	marshaledPod, err := json.Marshal(pod)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+    return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
+    
+    //BLOCK B
+    if FAILURE_CONDITION {
+		return admission.Denied(fmt.Sprintf("reason for failure"))
+	}
+
+    return admission.Allowed("")
+    
+    //END BRANCH
+}
+
+// podInteraction implements inject.Client.
+// A client will be automatically injected.
+
+// InjectClient injects the client.
+func (i *podInteraction) InjectClient(c client.Client) error {
+	i.client = c
+	return nil
+}
+
+// podInteraction implements admission.DecoderInjector.
+// A decoder will be automatically injected.
+
+// InjectDecoder injects the decoder.
+func (i *podInteraction) InjectDecoder(d *admission.Decoder) error {
+	i.decoder = d
+	return nil
+}
+
+```
+
+
+### User facing usage (if needed)
+
+My suggested method for interacting with this feature is to have a command in the osdk which can be run after generating the base operator. The new command `$ operator-sdk generate webhook` will write the files cmd/manager/osdk-webhook.go, cmd/manager/mutationwebhook.go, and cmd/manager/validationwebhook.go
+
+Flags
+
+* `--port` int - Port to start webhook server on
+* `--cert-dir` string - Certificate directory
+
+### Observations and open questions
+
+[Here](https://github.com/operator-framework/operator-sdk-samples/pull/63) is an example of the planned implementation on the alpha version of controller runtime (out of date, currently being re-implemented with new/updated design). 
```

Comment:
As this is the proposal, not sure the out of date part belongs here. You can just leave a comment in the PR for that note.